plugin/evm/vm.go

```diff
@@ -277,6 +279,7 @@ func (vm *VM) Initialize(
 	vm.db = versiondb.New(baseDB)
 	vm.acceptedBlockDB = prefixdb.New(acceptedPrefix, vm.db)
 	vm.metadataDB = prefixdb.New(metadataPrefix, vm.db)
+	vm.precompileBackend = precompile.NewNoopBackend()
```

Comment:
Can we pass the actual backend here instead of `NewNoopBackend` ?

I think we should create a mock backend that we can use for testing instead of a noop backend. The difference here would be that a mock backend would allow us to track what operations are performed on it for validation during tests, whereas a no-op backend is just no-ops.

We can specify the actual one instead of a noop backend here, because the warp precompile itself will not be merged until it is ready for production.

Reply:
Passing in initialized warp backend to make `precompileBackend`, and created a mock backend with stats for tracking. 

Looking to see how/where I can have a test that creates a new blockchain with `MockBackend`, submits transaction to warp precompile to emit log, blockchain accepts block and calls `handlePrecompilePostAccept`, warp precompile `OnAccept` gets called, check stats for backend.